source/portfolio/evolutionary_algorithm.cpp

```diff
@@ -0,0 +1,1002 @@
+    }
+    enum evolutionary_algorithm::island_structure
+    evolutionary_algorithm::island_structure() {
+        return this->island_structure_;
+    }
+    void evolutionary_algorithm::island_structure(
+        enum evolutionary_algorithm::island_structure value) {
+        this->island_structure_ = value;
+    }
+    enum evolutionary_algorithm::island_migration_policy
+    evolutionary_algorithm::island_migration_policy() {
+        return this->island_migration_policy_;
+    }
+    void evolutionary_algorithm::island_migration_policy(
+        enum evolutionary_algorithm::island_migration_policy value) {
+        this->island_migration_policy_ = value;
+    }
+    enum evolutionary_algorithm::island_replacement_policy
+    evolutionary_algorithm::island_replacement_policy() {
+        return this->island_replacement_policy_;
+    }
+    void evolutionary_algorithm::island_replacement_policy(
+        enum evolutionary_algorithm::island_replacement_policy value) {
+        this->island_replacement_policy_ = value;
+    }
+    size_t evolutionary_algorithm::migration_epoch() {
+        return this->migration_epoch_;
+    }
+    void evolutionary_algorithm::migration_epoch(size_t value) {
+        this->migration_epoch_ = value;
+    }
+    size_t evolutionary_algorithm::migration_size() {
+        return this->migration_size_;
+    }
+    void evolutionary_algorithm::migration_size(size_t value) {
+        this->migration_size_ = value;
+    }
+    double evolutionary_algorithm::fitness_sharing_niche_size() {
+        return fitness_sharing_niche_size_;
+    }
+    void evolutionary_algorithm::fitness_sharing_niche_size(double value) {
+        this->fitness_sharing_niche_size_ = value;
+    }
+    size_t evolutionary_algorithm::max_generations() {
+        return this->max_generations_;
+    }
+    void evolutionary_algorithm::max_generations(size_t value) {
+        this->max_generations_ = value;
+    }
+    double evolutionary_algorithm::children_proportion() {
+        return this->children_proportion_;
+    }
+    void evolutionary_algorithm::children_proportion(double value) {
+        this->children_proportion_ = value;
+    }
+    double evolutionary_algorithm::crossover_probability() {
+        return this->crossover_probability_;
+    }
+    void evolutionary_algorithm::crossover_probability(double value) {
+        this->crossover_probability_ = value;
+    }
+    double evolutionary_algorithm::mutation_strength() {
+        return this->mutation_strength_;
+    }
+    void evolutionary_algorithm::mutation_strength(double value) {
+        this->mutation_strength_ = value;
+    }
+    bool evolutionary_algorithm::competition_between_parents_and_children() {
+        return this->competition_between_parents_and_children_;
+    }
+    void evolutionary_algorithm::competition_between_parents_and_children(
+        bool value) {
+        this->competition_between_parents_and_children_ = value;
```

Comment:
Nos testes iniciais, quando for multiobjetivo, é bom que isso seja verdadeiro. Em mono, melhor ser falso.